securedrop_client/storage.py

```diff
@@ -329,10 +390,15 @@ def __update_submissions(
     # The uuids remaining in local_uuids do not exist on the remote server, so
     # delete the related records.
     for deleted_submission in local_submissions_by_uuid.values():
-        delete_single_submission_or_reply_on_disk(deleted_submission, data_dir)
-        session.delete(deleted_submission)
-        logger.debug(f"Deleted {model.__name__} {deleted_submission.uuid}")
 
+        # The local method could have deleted these files and submissions already
+        try:
+            delete_single_submission_or_reply_on_disk(deleted_submission, data_dir)
+            session.delete(deleted_submission)
+            logger.debug(f"Deleted {model.__name__} {deleted_submission.uuid}")
+        except NoResultFound:
+            logger.info(f"Tried to delete submission {deleted_submission.uuid}, but "
```

Comment:
This isn't a `warning` or `error` situation; when conversations are deleted _via the client_, it would be expected that our local delete would delete submissions before the sync-delete can do so.